File: C/Algorithms/684.c
// Note: For UnionFind structure see Structures/UnionFind/uf.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int *findRedundantConnection(int **edges, int edgesSize, int *edgesColSize,
                             int *returnSize) {
  struct UnionFind *uf = newUnionFind(0, edgesSize + 1);

  int lasts = 0, lastd = 0;
  for (int i = 0; i < edgesSize; i++) {
    int s = edges[i][0], d = edges[i][1];

    if (Find(uf, s) == Find(uf, d)) {
      lasts = s;
      lastd = d;
    } else {
      Union(uf, s, d);
    }
  }
  // We've found last source and last destination:
  *returnSize = 2;
  int *res = malloc(*returnSize * sizeof(*res));
  res[0] = lasts;
  res[1] = lastd;

  UnionFindFree(uf);
  return res;
}

File: Rust/Algorithms/684.rs
// Note: For UnionFind structure see Structures/UnionFind/uf.rs
impl Solution {
    pub fn find_redundant_connection(edges: Vec<Vec<i32>>) -> Vec<i32> {
        let mut uf = UnionFind::with_elements(0, edges.len() + 1);
        let mut last = vec![];
        for e in edges {
            let (a, b) = (e[0] as usize, e[1] as usize);
            if uf.find(a) == uf.find(b) {
                last = vec![a as i32, b as i32];
            } else {
                uf.union(a, b);
            }
        }
        last
    }
}
